File: CS/WindowsFormsApplication3/InplaceDataNavigator/MyDataNavigator.cs
using DevExpress.Utils.Drawing;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.ViewInfo;
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Windows.Forms;

namespace WindowsFormsApplication3
{
    public class MyDataNavigator : DataNavigator
    {
        protected override BaseStyleControlViewInfo CreateViewInfo()
        {
            return new MyNavigatorControlViewInfo(this);
        }

        public void Draw(PaintEventArgs e)
        {
            (ViewInfo as MyNavigatorControlViewInfo).CalcViewInfo(e.Graphics, MouseButtons.None, new Point(-6000, -6000), e.ClipRectangle);
            GraphicsCache cache = new GraphicsCache(new DXPaintEventArgs(e));
            try
            {
                ControlGraphicsInfoArgs info = new ControlGraphicsInfoArgs(ViewInfo, cache, ViewInfo.Bounds);
                ISupportGlassRegions reg = Parent as ISupportGlassRegions;
                if (reg != null)
                {
                    info.IsDrawOnGlass = reg.IsOnGlass(Parent.RectangleToClient(RectangleToScreen(ViewInfo.Bounds)));
                }
                Painter.Draw(info);
                RaisePaintEvent(this, e);
            }
            finally
            {
                cache.Dispose();
            }
        }

        class MyNavigatorControlViewInfo : NavigatorControlViewInfo
        {
            public MyNavigatorControlViewInfo(BaseStyleControl owner) : base(owner) { }

            public new void CalcViewInfo(Graphics g, MouseButtons buttons, Point p, Rectangle rec)
            {
                base.CalcViewInfo(g, buttons, p, rec);
                this.Bounds = rec;
                CalcRects();
            }
        }
    }
}

File: VB/WindowsFormsApplication3/InplaceDataNavigator/MyDataNavigator.vb
Imports Microsoft.VisualBasic
Imports DevExpress.Utils.Drawing
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.ViewInfo
Imports System
Imports System.Collections.Generic
Imports System.Drawing
Imports System.Linq
Imports System.Windows.Forms

Namespace WindowsFormsApplication3
	Public Class MyDataNavigator
		Inherits DataNavigator
		Protected Overrides Function CreateViewInfo() As BaseStyleControlViewInfo
			Return New MyNavigatorControlViewInfo(Me)
		End Function

		Public Sub Draw(ByVal e As PaintEventArgs)
			TryCast(ViewInfo, MyNavigatorControlViewInfo).CalcViewInfo(e.Graphics, MouseButtons.None, New Point(-6000, -6000), e.ClipRectangle)
			Dim cache As New GraphicsCache(New DXPaintEventArgs(e))
			Try
				Dim info As New ControlGraphicsInfoArgs(ViewInfo, cache, ViewInfo.Bounds)
				Dim reg As ISupportGlassRegions = TryCast(Parent, ISupportGlassRegions)
				If reg IsNot Nothing Then
					info.IsDrawOnGlass = reg.IsOnGlass(Parent.RectangleToClient(RectangleToScreen(ViewInfo.Bounds)))
				End If
				Painter.Draw(info)
				RaisePaintEvent(Me, e)
			Finally
				cache.Dispose()
			End Try
		End Sub

		Private Class MyNavigatorControlViewInfo
			Inherits NavigatorControlViewInfo
			Public Sub New(ByVal owner As BaseStyleControl)
				MyBase.New(owner)
			End Sub

			Public Shadows Sub CalcViewInfo(ByVal g As Graphics, ByVal buttons As MouseButtons, ByVal p As Point, ByVal rec As Rectangle)
				MyBase.CalcViewInfo(g, buttons, p, rec)
				Me.Bounds = rec
				CalcRects()
			End Sub
		End Class
	End Class
End Namespace
